pymc3/variational/advi.py

```diff
@@ -149,6 +166,27 @@ def advi(vars=None, start=None, model=None, n=5000, accurate_elbo=False,
             elif i % (n // 10) == 0 and i > 0:
                 avg_elbo = elbos[i - n // 10:i].mean()
                 progress.set_description('Average ELBO = {:,.5g}'.format(avg_elbo))
+
+            if i % eval_elbo == 0:
+                elbo_prev = elbo_current
+                elbo_current = elbos[i]
+                delta_elbo = abs((elbo_current - elbo_prev) / elbo_prev)
+                circ_buff.append(delta_elbo)
+                avg_delta = np.mean(circ_buff)
+                med_delta = np.median(circ_buff)
+
+                if avg_delta < tol_obj:
+                    pm._log.info('Mean ELBO converged.')
+                    converged = True
+                    elbos = elbos[:(i + 1)]
```

Comment:
Second guessing this change, the slightly nasty indexing there isn't really as bad as all the verboseness that gets added due to deque not being able to handle slice operations. Right now that involves something like the following:

```
avg_elbo = np.fromiter(islice(elbos, i - n // 10, i), float).mean()
```